scarlet/display.py

```diff
@@ -321,74 +355,76 @@ def show_sources(
             observation is not None
         ), "Provide matched observation to show observed frame"
 
-    panels = 1 + sum((show_observed, show_rendered, show_sed))
+    panels = sum((show_model, show_observed, show_rendered, show_sed))
     if figsize is None:
         figsize = (3 * panels, 3 * len(list(sources)))
     fig, ax = plt.subplots(len(list(sources)), panels, figsize=figsize)
     for k, src in enumerate(sources):
-
         if hasattr(src, "center"):
-            center = np.array(src.center)
-            # center in src bbox coordinates
-            if src.bbox is not None:
-                center_ = center - np.array(src.bbox.origin[1:])
-            else:
-                center_ = center
-            # center in observed coordinates
-            center__ = center - np.array(src.frame.origin[1:])
+            center = np.array(src.center)[::-1]
         else:
             center = None
 
         panel = 0
-        frame_ = src.frame
-        src.set_frame(src.bbox)
+
+        model = src.get_model()
         if isinstance(src, ComponentTree):
-            model = 0
             seds = []
             for component in src:
                 model_ = component.get_model()
                 seds.append(model_.sum(axis=(1, 2)))
-                model += model_
         else:
-            model = src.get_model()
             seds = [model.sum(axis=(1, 2))]
-        src.set_frame(frame_)
-        ax[k][panel].imshow(img_to_rgb(model, norm=norm, channel_map=channel_map, mask=model.sum(axis=0)==0))
-        ax[k][panel].set_title("Model Source {}".format(k))
-        if center is not None:
-            ax[k][panel].plot(*center_[::-1], "wx", mew=1, ms=10)
 
-        if show_rendered:
+        if use_mask:
+            mask = model.sum(axis=0) == 0
+        else:
+            mask = None
+
+        extent = get_extent(src.bbox)
+        rendered_box = src.bbox
+        if show_model:
+            # Show the unrendered model in it's bbox
+            ax[k][panel].imshow(
+                img_to_rgb(model, norm=norm, channel_map=channel_map, mask=mask), extent=extent)
+            ax[k][panel].set_title("Model Source {}".format(k))
+            ax[k][panel].set_xlim(extent[0], extent[1])
+            ax[k][panel].set_ylim(extent[2], extent[3])
+            if center is not None and mark_centers:
+                ax[k][panel].plot(*center, "wx", mew=1, ms=10)
             panel += 1
-            model = src.get_model()
-            model = observation.render(model)
-            ax[k][panel].imshow(img_to_rgb(model, norm=norm, channel_map=channel_map))
+
+        if show_rendered:
+            # Center and show the rendered model
+            model = src.project(frame=rendered_box)
+            model = observation.render(model, in_frame=False)
+            ax[k][panel].imshow(img_to_rgb(model, norm=norm, channel_map=channel_map), extent=extent)
```

Comment:
but then all panels need to use masks!

Reply:
see above, this is a different mask that I use for troubleshooting.

Reply:
yes, and you still can. but most people will apply only observational masks and expect consistency.

Reply:
See previous post. I changed the boolean `use_mask=False` to `model_mask=None`, which now allows the user to specify a mask for the model. It doesn't make sense to apply the same mask to the rendered image because we would have to create a new "convolved mask" in the rendered space to make this consistent, which would probably be more confusing.